### PR TITLE
Externalizing SAM site locations

### PR DESCRIPTION
**Pull request by ja2-derek**
This PR externalizes SAM site locations. The goal is to allow map mods to override SAM site locations. Though the SAM site list is now a `std::vector`, adding or removing SAM sites are not allowed.

See also #1011 and #665.

### Variables converted

- `pSamList[]`:  Now dynamically defined by JSON
- `pSamGridNoAList[]`:  This is the "anchor" tile of the SAM computer terminal. This is always the lower tile (the one with higher gridNo)
- `pSamGridNoBList[]`:  The other tile of the SAM computer
- `gbSAMGraphicList[]`:  It is now derived from SAM gridNos

### Variables not changed
- `fSamSiteFound[]`  Not changing for save compatibility

### Restrictions

Restrictions enforced in `SamSiteModel.cc`.

- Must define exactly 4 SAM sites
    - Breaks Saves, if more than 4 SAM sites
    - Breaks hard-coded references from Skyrider and Meanwhile dialogues, if fewer than 4 SAMs
- SAM gridNos must be 2 adjacent tiles

### What's not covered

- Airspace control table
- Hard-coded references to SAM sectors, such as `HandleSkyRiderMonologueAboutDrassenSAMSite()`, `ChooseSuitableGarrisonToProvideReinforcements()`
- Land type strings. SAM sectors (D2, I8, etc) are hard-coded-mapped to respective land type strings. It is another big chunk of work to externalize sector land type mappings. The land type display will be wrong if modder moved the SAMs sites to other sectors.
- A section of code in Assignments.cc commented out, saying: `/* No point in allowing SAM site repair any more.  Jan/13/99.  ARM` - _**Should all these be deleted?**_

### Testing

Manually tested in game.

1. Recruited Skyrider and he told us about Drassen SAM site
1. Verified other SAM sectors are not named as SAM sites
1. Took over Drassen SAM site.
1. Skyrider told us about other SAM sites. Those sectors are now named SAM sites.
1. Damaged SAM computer. SAM computer looks damaged. Confirm `bSAMCondition` reduced via logs.
1. Queen's Army re-took Drassen SAM site
1. Re-took Drassen SAM site. Confirmed the SAM computer in good condition.

### CODE REVIEW

File: src/externalized/ContentManager.h [hunk 1]
```diff
@@ -13,14 +13,16 @@
...
 
 
+class BloodCatPlacementsModel;
+class BloodCatSpawnsModel;
 class DealerInventory;
 class GamePolicy;
 class IMPPolicy;
-class BloodCatPlacementsModel;
-class BloodCatSpawnsModel;
 class TownModel;
```
Comment (lynxlynxlynx) [inline src/externalized/ContentManager.h:21]:
```suggestion

```

File: src/externalized/DefaultContentManager.cc [hunk 2]
```diff
@@ -1068,26 +1071,39 @@ bool DefaultContentManager::loadStrategicLayerData() {
...
 	m_movementCosts = MovementCostsModel::deserialize(*json);
+	delete json;
 
 	json = readJsonDataFile("strategic-map-npc-placements.json");
 	for (auto& element : json->GetArray())
 	{
 		auto placement = NpcPlacementModel::deserialize(element);
 		m_npcPlacements.insert(std::make_pair(placement->profileId, placement));
 	}
+	delete json;
```
Comment (lynxlynxlynx) [inline src/externalized/DefaultContentManager.cc:1106]: most of these deletes don't belong in this PR; they're separate fixes.

Comment (ja2-derek) [reply src/externalized/DefaultContentManager.cc:1106]: I will rewrite the commits and force-push, with the review comments and without these.

**Comment by ja2-derek** [comment]
Re-wrote the commits, with the PR suggestions included

**Comment by lynxlynxlynx** [comment]
Thanks!
